cmd/csi-node-driver-registrar/node_register.go

```diff
@@ -77,3 +80,20 @@ func nodeRegister(
 	// If gRPC server is gracefully shutdown, exit
 	os.Exit(0)
 }
+
+func buildSocketPath(csiDriverName string) string {
+	return fmt.Sprintf("/registration/%s-reg.sock", csiDriverName)
+}
+
+func removeRegSocket(csiDriverName string) {
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGTERM)
+	<-sigc
+	socketPath := buildSocketPath(csiDriverName)
+	err := os.Remove(socketPath)
```

Comment:
In the main code path we remove it only if it's a socket (not sure why). Can you please clarify why we are removing in this path without doing the same check ? I think we should have a single code path for delete which is called from the sync path and from the async signal path as well.

Reply:
the check is not required here, only we are going to delete the socket if it is not present we are discarding the error, we are good here

Reply:
The original check seems wrong (if the file isn't a socket, proceed and listen to it anyway...?) It should be fixed but doesn't have to be in this PR

I agree we don't need the check here - the socket should've been created by a node-driver-registrar and if something dropped a different file in its place, it did so at its own risk